README.md

```diff
@@ -18,40 +20,48 @@ The different parts of Cloudflow work in unison to dramatically accelerate your
 
 ## Basic Concepts
 
-// TODO Add an illustration here
+<p>
+<img src="./docs/images/apps.png" width="600">
+
+<i>Basic components of a Cloudflow Application</i>
+</p>
 
 As we discussed above, Cloudflow allows developers to quickly build and deploy distributed stream processing applications by breaking such applications into smaller stream processing units called `Streamlets`.
 Each `Streamlet` represents an independent stream processing component that implements a self-contained stage of the application logic.
 `Streamlets` let you break down your application into logical pieces that communicate with each other in a streaming fashion to accomplish an end to end goal.
 
-`Streamlets` can be composed into larger systems using _blueprints_, which specify how `Streamlets` can be connected together to form a topology.
+`Streamlets` can be composed into larger systems using _blueprints_, which specify how `Streamlets` can be connected together to form a topology. The topology formed defines a DAG.
 
-`Streamlets` expose one or more inlets and outlets that represent the data consumed and produced by the `Streamlet`, respectively. 
-Inlets and outlets are schema-driven, ensuring that data flows are always consistent and that connections between `streamlets` are compatible. 
+`Streamlets` expose one or more inlets and outlets that represent the data consumed and produced by the `Streamlet`, respectively.
+Inlets and outlets are schema-driven, ensuring that data flows are always consistent and that connections between `streamlets` are compatible.
+In the diagram above *Streamlet 1* has one outlet which feeds data to *Streamlet 2* inlet. *Streamlet 1* is a component that generates data or could get its data from an external system eg. via an http request.
+Then *Streamlet 2* outlet feeds its data output to *Streamlet 3* inlet. *Streamlet 2* in this application does the actual data processing.
+*Streamlet 3* then may store its data to some external system or just print to console.
```

Comment:
```suggestion
*Streamlet 3* then may store its data to some external system.
```

Reply:
(just printing is not that useful)

Reply:
it is for the dev lifecycle isn't?

Reply:
you can print anywhere, you don't have to create a streamlet just for printing, is what I mean.